torchgeometry/image/laplacian.py

```diff
@@ -0,0 +1,175 @@
+from typing import Tuple
+
+import math
+import torch
+import torch.nn as nn
+from torch.nn.functional import conv2d
+
+
+def laplacian_1d(window_size) -> torch.Tensor:
+    r"""One could also use the Laplacian of Gaussian formula
+        to design the filter.
+    """
+
+    filter_1d = torch.ones(window_size)
+    filter_1d[int(math.floor(window_size / 2))] = 1 - window_size
+    laplacian_1d: torch.Tensor = filter_1d
+    return laplacian_1d
+
+
+def get_laplacian_kernel(kernel_size: int) -> torch.Tensor:
+    r"""Function that returns the coefficients of a 1D Laplacian filter
+
+    Args:
+        kernel_size (int): filter size. It should be odd and positive.
+
+    Returns:
+        Tensor (float): 1D tensor with laplacian filter coefficients.
+
+    Shape:
+        - Output: math:`(ksize, 0)`
+
+    Examples::
+        >>> tgm.image.get_laplacian_kernel(3)
+        tensor([ 1., -2.,  1.])
+
+        >>> tgm.image.get_laplacian_kernel(5)
+        tensor([ 1.,  1., -4.,  1.,  1.])
+
+    """
+    if not isinstance(kernel_size, int) or kernel_size % 2 == 0 or \
+            kernel_size <= 0:
+        raise TypeError("ksize must be an odd positive integer. Got {}"
+                        .format(kernel_size))
+    window_1d: torch.Tensor = laplacian_1d(kernel_size)
+    return window_1d
+
+
+def get_laplacian_kernel2d(kernel_size: int) -> torch.Tensor:
+    r"""Function that returns Gaussian filter matrix coefficients.
+
+    Args:
+        ksize (int): filter size should be odd.
+
+    Returns:
+        Tensor: 2D tensor with laplacian filter matrix coefficients.
+
+    Shape:
+        - Output: :math:`(ksize, ksize)`
+
+    Examples::
+
+        >>> tgm.image.get_laplacian_kernel2d(3)
+        tensor([[ 1.,  1.,  1.],
+        [ 1., -8.,  1.],
+        [ 1.,  1.,  1.]])
+
+        >>> tgm.image.get_laplacian_kernel2d(5)
+        tensor([[  1.,   1.,   1.,   1.,   1.],
+        [  1.,   1.,   1.,   1.,   1.],
+        [  1.,   1., -24.,   1.,   1.],
+        [  1.,   1.,   1.,   1.,   1.],
+        [  1.,   1.,   1.,   1.,   1.]])
+
+    """
+    if not isinstance(kernel_size, int) or kernel_size % 2 == 0 or \
+            kernel_size <= 0:
+        raise TypeError("ksize must be an odd positive integer. Got {}"
+                        .format(kernel_size))
+
+    kernel = torch.ones((kernel_size, kernel_size))
+    mid = int(math.floor((kernel_size / 2)))
```

Comment:
torch.floor, torch.pow

Reply:
These functions apply to tensors. What I want is just an int.

Reply:
Though I can convert it to introduce. So would change

Reply:
oh wait. just do: kernel_size // 2